### PR TITLE
add schema version to batch write client and server

### PR DESCRIPTION
**Pull request by reyreaud-l**
### What's being changed:

* Update remote index batch write client with schema version
* Update replication client batch write with schema version
* Update remote index server batch write with schema version

### Review checklist

- [ ] Documentation has been updated, if necessary. Link to changed documentation:
- [ ] Chaos pipeline run or not necessary. Link to pipeline:
- [ ] All new code is covered by tests where it is reasonable.
- [ ] Performance tests have been run or not necessary.

### CODE REVIEW

File: usecases/classification/writer.go [hunk 1]
```diff
@@ -127,7 +127,7 @@ func (r *batchWriter) saveObjects(items objects.BatchObjects) {
 	defer cancel()
 
 	if len(items) > 0 {
-		saved, err := r.vectorRepo.BatchPutObjects(ctx, items, nil)
+		saved, err := r.vectorRepo.BatchPutObjects(ctx, items, nil, 0)
```
Comment (redouan-rhazouani) [inline usecases/classification/writer.go:130]: Do we need to include the version here?

File: usecases/objects/batch_add.go [hunk 2]
```diff
@@ -61,7 +61,7 @@ func (b *BatchManager) AddObjects(ctx context.Context, principal *models.Princip
 
 	beforePersistence := time.Now()
 	defer b.metrics.BatchOp("total_persistence_level", beforePersistence.UnixNano())
-	if res, err = b.vectorRepo.BatchPutObjects(ctx, batchObjects, repl); err != nil {
+	if res, err = b.vectorRepo.BatchPutObjects(ctx, batchObjects, repl, 0); err != nil {
```
Comment (redouan-rhazouani) [inline usecases/objects/batch_add.go:64]: Do we need to include the version here?

File: usecases/classification/writer.go (hunk 1 above)
Comment (reyreaud-l) [reply usecases/classification/writer.go:130]: Discussed orally: Will check with core team to understand that usage.

File: adapters/handlers/rest/clusterapi/indices.go [hunk 3]
```diff
@@ -364,7 +365,14 @@ func (i *indices) postObjectBatch(w http.ResponseWriter, r *http.Request,
 		return
 	}
 
-	errs := index.IncomingBatchPutObjects(r.Context(), shard, objs)
+	var schemaVersion uint64
```
Comment (redouan-rhazouani) [inline adapters/handlers/rest/clusterapi/indices.go:368]: Could you refactor the code for reading the version into a function to make it reusable

**Pull request merged**